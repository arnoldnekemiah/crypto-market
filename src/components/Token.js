import { useSelector } from 'react-redux';
import { NavLink } from 'react-router-dom';
import PropTypes from 'prop-types';
import { BsChevronLeft, BsFillMicFill } from 'react-icons/bs';
import { IoSettingsSharp } from 'react-icons/io5';

const Token = ({ rank }) => {
  const { cryptoArray } = useSelector((store) => store.crypto);

  const token = cryptoArray.filter((coin) => coin.rank === rank)[0];

  return (
    <article>
      <header className="row">
        <div className="homeLink first">
          <NavLink className="link" to="/" data-testid="navLink">
            <BsChevronLeft />
          </NavLink>
        </div>
        <p>Stats</p>
        <div className="row dummyIcons">
          <BsFillMicFill />
          <IoSettingsSharp />
        </div>
      </header>
      <div>
        <div className="statsTitle row">
          <h1>{`#${token.rank}`}</h1>
          <h2 className="column coinName">
            <span>{`${token.name}`}</span>
            <span>{`(${token.symbol})`}</span>
          </h2>
        </div>
        <div className="coinStats">
          <p className="row stats">
            <span>Market Cap</span>
            <span>{`$ ${token.marketCapUsd}`}</span>
          </p>
          <p className="row stats">
            <span>Price</span>
            <span>{`$ ${token.priceUsd}`}</span>
          </p>
          <p className="row stats">
            <span>Volume traded over past 24h</span>
            <span>{`$ ${token.volumeUsd24Hr}`}</span>
          </p>
          <p className="row stats">
            <span>Supply</span>
            <span>{`${token.supply} units`}</span>
          </p>
          <p className="row stats">
            <span>Max-supply</span>
            <span>{`${token.maxSupply} units`}</span>
          </p>
        </div>
      </div>
    </article>
  );
};

Token.propTypes = {
  rank: PropTypes.string.isRequired,
};
export default Token;
