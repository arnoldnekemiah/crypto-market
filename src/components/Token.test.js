import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom'; // Import BrowserRouter
import configureStore from 'redux-mock-store';
import Token from './Token';

// Create a mock Redux store
const mockStore = configureStore([]);

describe('Token', () => {
  it('renders correctly', () => {
    const initialState = {
      crypto: {
        cryptoArray: [
          {
            rank: '1',
            name: 'Bitcoin',

          },

        ],
      },

    };

    const store = mockStore(initialState);

    const component = renderer.create(
      <Provider store={store}>
        <Router>
          <Token rank="1" />
        </Router>
      </Provider>,
    );

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
