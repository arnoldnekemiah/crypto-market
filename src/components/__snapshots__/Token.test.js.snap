// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Token renders correctly 1`] = `
<article>
  <header
    className="row"
  >
    <div
      className="homeLink first"
    >
      <a
        aria-current="page"
        className="link active"
        data-testid="navLink"
        href="/"
        onClick={[Function]}
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 16 16"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M11.354 1.646a.5.5 0 0 1 0 .708L5.707 8l5.647 5.646a.5.5 0 0 1-.708.708l-6-6a.5.5 0 0 1 0-.708l6-6a.5.5 0 0 1 .708 0z"
            fillRule="evenodd"
          />
        </svg>
      </a>
    </div>
    <p>
      Stats
    </p>
    <div
      className="row dummyIcons"
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 16 16"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M5 3a3 3 0 0 1 6 0v5a3 3 0 0 1-6 0V3z"
        />
        <path
          d="M3.5 6.5A.5.5 0 0 1 4 7v1a4 4 0 0 0 8 0V7a.5.5 0 0 1 1 0v1a5 5 0 0 1-4.5 4.975V15h3a.5.5 0 0 1 0 1h-7a.5.5 0 0 1 0-1h3v-2.025A5 5 0 0 1 3 8V7a.5.5 0 0 1 .5-.5z"
        />
      </svg>
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 512 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M256 176a80 80 0 1080 80 80.24 80.24 0 00-80-80zm172.72 80a165.53 165.53 0 01-1.64 22.34l48.69 38.12a11.59 11.59 0 012.63 14.78l-46.06 79.52a11.64 11.64 0 01-14.14 4.93l-57.25-23a176.56 176.56 0 01-38.82 22.67l-8.56 60.78a11.93 11.93 0 01-11.51 9.86h-92.12a12 12 0 01-11.51-9.53l-8.56-60.78A169.3 169.3 0 01151.05 393L93.8 416a11.64 11.64 0 01-14.14-4.92L33.6 331.57a11.59 11.59 0 012.63-14.78l48.69-38.12A174.58 174.58 0 0183.28 256a165.53 165.53 0 011.64-22.34l-48.69-38.12a11.59 11.59 0 01-2.63-14.78l46.06-79.52a11.64 11.64 0 0114.14-4.93l57.25 23a176.56 176.56 0 0138.82-22.67l8.56-60.78A11.93 11.93 0 01209.94 26h92.12a12 12 0 0111.51 9.53l8.56 60.78A169.3 169.3 0 01361 119l57.2-23a11.64 11.64 0 0114.14 4.92l46.06 79.52a11.59 11.59 0 01-2.63 14.78l-48.69 38.12a174.58 174.58 0 011.64 22.66z"
        />
      </svg>
    </div>
  </header>
  <div>
    <div
      className="statsTitle row"
    >
      <h1>
        #1
      </h1>
      <h2
        className="column coinName"
      >
        <span>
          Bitcoin
        </span>
        <span>
          (undefined)
        </span>
      </h2>
    </div>
    <div
      className="coinStats"
    >
      <p
        className="row stats"
      >
        <span>
          Market Cap
        </span>
        <span>
          $ undefined
        </span>
      </p>
      <p
        className="row stats"
      >
        <span>
          Price
        </span>
        <span>
          $ undefined
        </span>
      </p>
      <p
        className="row stats"
      >
        <span>
          Volume traded over past 24h
        </span>
        <span>
          $ undefined
        </span>
      </p>
      <p
        className="row stats"
      >
        <span>
          Supply
        </span>
        <span>
          undefined units
        </span>
      </p>
      <p
        className="row stats"
      >
        <span>
          Max-supply
        </span>
        <span>
          undefined units
        </span>
      </p>
    </div>
  </div>
</article>
`;
